snapcraft/tests/test_commands_snap.py

```diff
@@ -40,16 +41,63 @@ class SnapCommandTestCase(tests.TestCase):
       plugin: nil
 """
 
+    def setUp(self):
+        super().setUp()
+
+        patcher = mock.patch('snapcraft.internal.lifecycle.Popen',
+                             new=mock.Mock(wraps=subprocess.Popen))
+        self.popen_mock = patcher.start()
```

Comment:
we can start being obnoxious about terminology here now. If you call it self.popen_spy then it will be a little clearer in the test that you are keeping the original behaviour and just recording the calls.